From d1b67c4ad4210a4d827d4ad8754e9b47b28f0658 Mon Sep 17 00:00:00 2001
From: Bram Vlerick <bram.vlerick@openpixelsystems.org>
Date: Mon, 17 Aug 2026 17:25:50 +0200
Subject: [PATCH 1/2] src/image.c: fix unused variable compilation error
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

n variable is defined and set but never actually used. This causes newer
compilers to generate the following error:

image.c: In function ‘image_hash_sha256’:
image.c:515:20: error: variable ‘n’ set but not used [-Werror=unused-but-set-variable=]
  515 |         int rc, i, n;
      |                    ^
image.c: In function ‘image_hash_sha256’:
image.c:515:20: error: variable ‘n’ set but not used [-Werror=unused-but-set-variable=]
  515 |         int rc, i, n;
      |                    ^
image.c: In function ‘image_hash_sha256’:
image.c:515:20: error: variable ‘n’ set but not used [-Werror=unused-but-set-variable=]
  515 |         int rc, i, n;
      |                    ^
image.c: In function ‘image_hash_sha256’:
image.c:515:20: error: variable ‘n’ set but not used [-Werror=unused-but-set-variable=]
  515 |         int rc, i, n;
      |                    ^
image.c: In function ‘image_hash_sha256’:
image.c:515:20: error: variable ‘n’ set but not used [-Werror=unused-but-set-variable=]
  515 |         int rc, i, n;
      |                    ^
image.c: In function ‘image_hash_sha256’:
image.c:515:20: error: variable ‘n’ set but not used [-Werror=unused-but-set-variable=]
  515 |         int rc, i, n;

This commit removes the n declaration and assignment.

Signed-off-by: Bram Vlerick <bram.vlerick@openpixelsystems.org>
---
 src/image.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/image.c b/src/image.c
index a828b5a..8da2901 100644
--- a/src/image.c
+++ b/src/image.c
@@ -512,17 +512,14 @@ int image_hash_sha256(struct image *image, uint8_t digest[])
 {
 	struct region *region;
 	SHA256_CTX ctx;
-	int rc, i, n;
+	int rc, i;
 
 	rc = SHA256_Init(&ctx);
 	if (!rc)
 		return -1;
 
-	n = 0;
-
 	for (i = 0; i < image->n_checksum_regions; i++) {
 		region = &image->checksum_regions[i];
-		n += region->size;
 #if 0
 		printf("sum region: 0x%04lx -> 0x%04lx [0x%04x bytes]\n",
 				region->data - image->buf,
-- 
2.55.0

